lib/jobs/message_latency_report.ex

```diff
@@ -210,13 +198,27 @@ defmodule Jobs.MessageLatencyReport do
     id
   end
 
-  defp get_percentile_row(percentile_index, rows_with_index) do
-    {percentile_row, _index} =
-      Enum.find(rows_with_index, {[], nil}, fn {_row, index} ->
-        index == percentile_index
-      end)
+  defp get_percentiles(rows_with_index) do
+    num_rows = Enum.count(rows_with_index)
+    percentile_95_index = calculate_percentile_index(num_rows, @percentile_95)
+    percentile_99_index = calculate_percentile_index(num_rows, @percentile_99)
 
-    percentile_row
+    {Enum.reduce_while(rows_with_index, {[], []}, fn {current_row, index},
+                                                     {percentile_95, percentile_99} ->
+       cond do
+         percentile_95 != [] and percentile_99 != [] ->
```

Comment:
nit: Since we know we're done at 99, could we halt there instead of doing one more loop?

Reply:
good point!